### PR TITLE
Jobs api

### LINKED ISSUE #388: [API] $.jobs
**$.jobs** API

https://help.sap.com/doc/3de842783af24336b6305a3c0223a369/2.0.03/en-US/$.jobs.Job.html

### PR DESCRIPTION
**Pull request by cheezus1**
Closes #388 

Tests will be added separately.

### CODE REVIEW

File: modules/api/api-xsjs/src/main/resources/META-INF/dirigible/xsk/api.js [hunk 1]
```diff
@@ -14,23 +14,24 @@
  */
 
 var $ = {};
-
+//
```
Comment (ThuF) [inline modules/api/api-xsjs/src/main/resources/META-INF/dirigible/xsk/api.js:17]: Not needed?

File: modules/engines/engine-xsjob/src/main/java/com/sap/xsk/xsjob/ds/scheduler/XSKSchedulerManager.java [hunk 2]
```diff
@@ -53,7 +54,9 @@ public static void scheduleJob(XSKJobDefinition jobDefinition) throws SchedulerE
           return;
         }
 
-        CronTrigger trigger = newTrigger().withIdentity(triggerKey).withSchedule(cronSchedule(jobDefinition.getCronExpression())).build();
+        CronTrigger trigger = newTrigger().withIdentity(triggerKey).startAt(new Date(jobDefinition.getStartAt().getTime()))
+            .endAt(new Date(jobDefinition.getEndAt().getTime()))
+            .withSchedule(cronSchedule(jobDefinition.getCronExpression()).withMisfireHandlingInstructionDoNothing()).build();
```
Comment (ThuF) [inline modules/engines/engine-xsjob/src/main/java/com/sap/xsk/xsjob/ds/scheduler/XSKSchedulerManager.java:59]: What will happen, if the `startTime` and/or the `endTime` is not configured?

**Review by ThuF — CHANGES_REQUESTED** [review]
Few minor comments. Please rebase on the `main` branch, as that will fix the build failures.

File: modules/engines/engine-xsjob/src/main/java/com/sap/xsk/xsjob/ds/scheduler/XSKSchedulerManager.java (hunk 2 above)
Comment (cheezus1) [reply modules/engines/engine-xsjob/src/main/java/com/sap/xsk/xsjob/ds/scheduler/XSKSchedulerManager.java]: We were only handling this case in the facade, we moved the check here as this method is called when the job synchronizer synchronizes automatically.

**Review by ThuF — APPROVED** [review]
+1

**Pull request merged**